atlasdb-client/src/main/java/com/palantir/atlasdb/cache/OffHeapTimestampCache.java

```diff
@@ -126,6 +149,7 @@ private static CacheDescriptor createNamespaceAndConstructCacheProposal(
         @Override
         public Map<Map.Entry<Long, Long>, Map.Entry<Long, Long>> apply(Set<Map.Entry<Long, Long>> request) {
             if (offHeapTimestampCache.cacheDescriptor.get().currentSize().get() >= offHeapTimestampCache.maxSize) {
+                offHeapTimestampCache.taggedMetricRegistry.meter(CACHE_NUKE).mark();
                 offHeapTimestampCache.clear();
```

Comment:
I think this needs to reset `CACHE_SIZE` to 0, or the metric should be something like `CACHE_WRITES` rather than `CACHE_SIZE`.

Reply:
Discussed off-line, since counter does not provide a reset we will write a new `Gauge` which can be reseted.

Reply:
Added a resettable counter implementation of a `Gauge`.